### DISCUSSION TITLE
PyCaret shouldn't apply the same pre-processing for every algorithm type

### DISCUSSION THREAD

**Discussion by KitTarzan** [pycaret/pycaret#3021] [Ideas]
I could be wrong here, but it seems to me that there is no way to prevent pycaret from applying some sort of categorical encoding to the features identified (either manually or automatically) as categorical. Ideally, there could be multiple (hidden) columns created by the setup function that PyCaret handles automatically so that when we have cat_features = ['size','colour'], these columns are converted to categorical pandas type (if needed), and the list is past to the appropriate categorical features parameter for LightGBM, XGBoost , and Catboost. These libraries all have ways of natively handling categorical variables which can be superior to the options pycaret provides for handling categorical labels. 

In the same vein, PyCaret could have multiple column versions for normalized/unnormalized. Since there are some algorithms that would benefit from the normalizing the features, or transforming the target (e.g. linear regression) and others that are less likely to (e.g. random forests), holding multiple preprocessed versions of the data would be helpful in this vein too. I may not want my extra trees regressor to use the normalized features, since I'd prefer to interpret the splits on the original scale for example. Since we can perform this workaround just by setting up a different experiments with different setup parameters and running different algorithms for each, I'm not as interested in this suggestion. But I am very keen in being able to handle categorical features with a bit more flexibility. As it is now, I can't truly use PyCaret in a full PoC workflow.
